pkg/transaction/logeventbuilder.go

```diff
@@ -300,11 +304,45 @@ func (b *transactionEventBuilder) SetProtocolDetail(protocolDetail interface{})
 	return b
 }
 
+func (b *transactionEventBuilder) SetRedactionConfig(config redaction.Redactions) EventBuilder {
+	if b.err != nil {
+		return b
+	}
+	b.redactionConfig = config
+	return b
+}
+
 func (b *transactionEventBuilder) Build() (*LogEvent, error) {
 	if b.err != nil {
 		return nil, b.err
 	}
 
+	// Set Target Path
+	if b.logEvent.UnRedactedTargetPath == "" {
+		return nil, errors.New("unredacted target path property not set in the LogEvent details")
+	}
+	if b.redactionConfig == nil {
+		b.logEvent.TargetPath, b.err = redaction.URIRedaction(b.logEvent.UnRedactedTargetPath)
+	} else {
+		b.logEvent.TargetPath, b.err = b.redactionConfig.URIRedaction(b.logEvent.UnRedactedTargetPath)
+	}
+	if b.err != nil {
+		return nil, b.err
+	}
+
+	//Set Resource Path
+	if b.logEvent.UnRedactedResourcePath == "" {
+		return nil, errors.New("unredacted resource path property not set in the LogEvent details")
```

Comment:
what if b.logEvent.UnRedactedResourcePath is set while b.logEvent.UnRedactedTargetPath is not, looks like b.logEvent.UnRedactedResourcePath would not be redacted